File: Interpreter/AsinFun.java
package calculator.xwg;

import java.util.LinkedList;

public class AsinFun extends CalculateFunction {
	String getName(){
		return "asin";
	}
	
	boolean execute(LinkedList<Complex> paraList, EvaluateContext context){
		if(paraList.size() != 1)
	    {
	        context.setErrorMessage(getName(), R.string.error_invalid_parameter_count);
	        return false;
	    }
	    Complex para = paraList.getFirst();
	    if(para.i != 0)
	    {
	        context.setErrorMessage(getName(), R.string.error_invalid_date_type);
	        return false;
	    }
	    context.setCurrentResult(new Complex(Math.asin(para.r)));
	    return checkResult(context);
	}
}

File: Interpreter/AsinFun.py
from Interpreter.CalculateFunction import *
from Interpreter.Complex import *

class AsinFun(CalculateFunction):
    def get_name(self):
        return 'asin'

    def execute(self, context,paraList):
        if (len(paraList) != 1):
            context.setErrorMessage(self.get_name(), R_string.error_invalid_parameter_count)
            return False

        para = paraList[0]
        if (para.i != 0):
            context.setErrorMessage(self.get_name(), R_string.error_invalid_date_type)
            return False

        context.setCurrentResult(Complex(math.asin(para.r)))
        return self.checkResult(context)
